iot_packet_verifier/src/burner.rs

```diff
@@ -85,33 +82,37 @@ where
         Ok(())
     }
 
-    pub async fn burn(&mut self) -> Result<(), BurnError<P::Error, S::Error>> {
+    pub async fn burn(&mut self) -> Result<(), BurnError<S::Error>> {
         // Create burn transaction and execute it:
 
-        let Some(Burn { payer, amount }) = self
-            .pending_burns
-            .fetch_next()
-            .await
-            .map_err(BurnError::SqlError)?
-        else {
+        let Some(Burn { payer, amount }) = self.pending_tables.fetch_next_burn().await? else {
             return Ok(());
         };
 
         tracing::info!(%amount, %payer, "Burning DC");
 
-        let amount = amount as u64;
-
+        let txn = self
+            .solana
+            .make_burn_transaction(&payer, amount)
+            .await
+            .map_err(BurnError::SolanaError)?;
+        self.pending_tables
+            .submit_txn(&payer, amount, txn.get_signature())
```

Comment:
if this is one of the places @bbalser mentioned where naming got tricky, i'd agree: `submit_txn` immediately followed by `submit_transaction`

Reply:
I renamed the pending tables methods to `add_pending_transaction` and `remove_pending_transaction`